### PR TITLE
bedtools: update to 2.28.0 (and remove patch)

### PR DESCRIPTION
**Pull request by arjanvandervelde**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by arjanvandervelde** [comment]
for reference:
https://github.com/arq5x/bedtools2/issues/705
https://github.com/macports/macports-ports/pull/3923

**Comment by arjanvandervelde** [comment]
@reneeotten @yan12125 here you go, this works. thanks for your time.

**Comment by yan12125** [comment]
> ok, I'm sorry for taking your time on this. the port compiles w/ clang after looking at this again. i've created a new pull request.

No problem! It's better to just force push to the original pull request.

My only remaining concern is about universal builds. @arjanvandervelde Have you checked if +universal builds universal binaries?

**Comment by arjanvandervelde** [comment]
> > ok, I'm sorry for taking your time on this. the port compiles w/ clang after looking at this again. i've created a new pull request.
> 
> No problem! It's better to just force push to the original pull request.
> 
> My only remaining concern is about universal builds. @arjanvandervelde Have you checked if +universal builds universal binaries?

ok, thanks. i checked it just now and it doesn't build universal. is that a show stopper? i have previously never checked for this and I'd be surprised if it worked in the past.

Error: bedtools cannot be installed for the configured universal_archs 'x86_64 i386' because it only supports the arch(s) ''.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port bedtools failed

**Comment by yan12125** [comment]
Looked into the history, the universal variant is added by another MacPorts member - https://trac.macports.org/ticket/41407#comment:2. This variant is not a requirement. As you're the maintainer, feel free to remove it if you believe no potential user needs it.

**Comment by arjanvandervelde** [comment]
> Looked into the history, the universal variant is added by another MacPorts member - https://trac.macports.org/ticket/41407#comment:2. This variant is not a requirement. As you're the maintainer, feel free to remove it if you believe no potential user needs it.

ok, i see. that's when i originally requested to have to port added. i'll remove the variant as i don't think there's a use case for that for these tools.

File: science/bedtools/Portfile [hunk 1]
```diff
@@ -18,25 +19,22 @@ long_description    The BEDTools utilities allow one to address common genomics
                     The utilities are largely based on four widely-used file formats: \
                     BED, GFF/GTF, VCF, and SAM/BAM.
 
-checksums           rmd160  2a0fc10ce07f6228ee006e297577560bc24e35af \
-                    sha256  eb7028247d0e4732f113076fdbd580b8601365f7830cf19649d4d3116af08dfd
+checksums           rmd160  0807fc84acd6d80f646ee387943ff6345203396a \
+                    sha256  511d8623038836ac8d1f9e260016867494b9040de65d0f0f1eb7feac7b698bbd \
```
Comment (mf2k) [inline science/bedtools/Portfile]: We are now adding the `size` parameter to `checksums`.

File: science/bedtools/Portfile [hunk 2]
```diff
@@ -18,25 +19,22 @@ long_description    The BEDTools utilities allow one to address common genomics
...
 
 depends_build       port:py27-sphinx
 
-depends_lib         port:zlib
+depends_lib         port:zlib port:lzma
```
Comment (mf2k) [inline science/bedtools/Portfile]: `depends_lib` should be changed to `depends_lib-append` to not overwrite portgroup dependencies.

File: science/bedtools/Portfile (hunk 1 above)
Comment (arjanvandervelde) [reply science/bedtools/Portfile]: ok, done. thanks.

File: science/bedtools/Portfile (hunk 2 above)
Comment (arjanvandervelde) [reply science/bedtools/Portfile]: ok, done. thanks.

**Comment by arjanvandervelde** [comment]
for reference: https://github.com/arq5x/bedtools2/issues/707

**Comment by yan12125** [comment]
LGTM, too. Thanks @mf2k for looking into details!

**Comment by arjanvandervelde** [comment]
Thanks!

**Comment by ryandesign** [comment]
> ok, thanks. i checked it just now and it doesn't build universal.

What version of macOS did you test on? You neglected to fill out that part of the pull request template. Note that macOS Mojave does not support building anything universal anymore, so an inability to compile something universal on Mojave would not be justification for removing universal support from a port.

> i have previously never checked for this and I'd be surprised if it worked in the past.

> Looked into the history, the universal variant is added by another MacPorts member - https://trac.macports.org/ticket/41407#comment:2.

Obviously I would not have added a universal variant without first verifying that it at least appeared to compile universal.

> i'll remove the variant as i don't think there's a use case for that for these tools.

It's unfortunate when people remove features that I spent time adding, especially without understanding why.

Some build systems automatically get the correct compiler and flags during the configure phase, but some build systems don't and need to be fed those values another way, such as in `build.args`. When you submitted the port, the bedtools build system apparently needed that extra help. At that time it was using C, and you submitted it with `build.args` that set `CC` and `CFLAGS`; at some point it switched to C++, setting `CXX` and `CXXFLAGS`.

You submitted it with flags that included `${configure.cc_archflags}`, which give the correct architecture flags to use for the C compiler when not building universal. I changed it to `[get_canonical_archflags]` which returns the correct list of architecture flags to use, regardless of whether the universal variant was selected.

You removed the part of `build.args` that set `CXXFLAGS="${configure.cxxflags} [get_canonical_archflags cxx]"`. So now no architecture flags are being used. You can see this in the [latest build log](https://build.macports.org/builders/ports-10.12_x86_64-builder/builds/88071/steps/install-port/logs/stdio) where you will note that the string `-arch x86_64` never appears. Also, because `${configure.cxxflags}` is no longer being supplied, the correct `-stdlib` flag isn't being given to the build system; since the port now requires C++11, this causes the build to faill on systems where the default C++ standard library doesn't support C++11, such as Mountain Lion and earlier.

Please reinstate the correct value for `CXXFLAGS` by restoring the lines you removed. Then you should also be able to reinstate the universal variant.

**Comment by ryandesign** [comment]
Actually we can't really see how the C++ compiler is being invoked by the build system; it's just printing lines like `  * compiling FileRecordMgr.cpp`. If there is a way to turn off these "silent rules" so that the full compile command is printed, that would be helpful.

I see the build system is now using both C and C++ compilers, so we should set both `CFLAGS` and `CXXFLAGS` to their correct values. You may also find that setting `LDFLAGS` is necessary; we won't know until we fix the earlier parts of the build.

**Comment by yan12125** [comment]
The universal variant is a nice feature, but recent changes in the Apple toolchain make ports with +universal difficult to maintain. It's far from trivial to build universal binaries with the latest Xcode, and buildbots do not support building non-default variants yet [1]. As a result it's infeasible for a port maintainer to check whether the universal variant are still working or not. Before the situation changes, I'm not against port maintainers dropping +universal support.

Other comments are still precious. @arjanvandervelde Could you have a look and see how to show build commands and use the correct c++ library? I guess build failures on Mac OS X 10.8 [2] is related to incorrect c++ library selection.

[1] https://trac.macports.org/ticket/52742
[2] https://build.macports.org/builders/ports-10.8_x86_64_legacy-builder/builds/90800/steps/install-port/logs/stdio

**Pull request merged**